app.rb

```diff
@@ -107,24 +104,20 @@ def redirect_pages
   end
 
   before do
-    begin
-      I18n.locale = dynamic_locale
-    rescue I18n::InvalidLocale
-      I18n.locale = I18n.default_locale
-    end
+    I18n.locale = dynamic_locale
+  rescue I18n::InvalidLocale
+    I18n.locale = I18n.default_locale
   end
 
   after '/:site/*' do
     # Any real page (starts with a site and doesn't end with an extension)
     # gets saved as the 'back' for the next pageload.
-    if sites.include?(params[:site]) && !request.fullpath.match(/\..+\z/)
-      cookies[:docs_back_path] = request.fullpath
-    end
+    cookies[:docs_back_path] = request.fullpath if sites.include?(params[:site]) && !request.fullpath.match(/\..+\z/)
```

Comment:
Since the conditions are on the longer side, I'd rather we stick to the block format to ensure readability.